examples/AdvancedOperations/AddDisplayUploadAd.php

```diff
@@ -0,0 +1,243 @@
+                'media_bundle_asset' => new MediaBundleAsset(
+                    [
+                        'data' => new BytesValue(['value' => $html5Zip]),
+                    ]
+                )
+            ]
+        );
+
+        // Creates an asset operation.
+        $assetOperation = new AssetOperation();
+        $assetOperation->setCreate($asset);
+
+        // Issues a mutate request to add the asset.
+        $assetServiceClient = $googleAdsClient->getAssetServiceClient();
+        $response = $assetServiceClient->mutateAssets(
+            $customerId,
+            [$assetOperation]
+        );
+
+        // Prints the resource name of the added media bundle asset.
+        /** @var MutateAssetResult $addedMediaBundleAsset */
+        $addedMediaBundleAsset = $response->getResults()[0];
+        printf(
+            "Uploaded media bundle asset with resource name: '%s'.%s",
+            $addedMediaBundleAsset->getResourceName(),
+            PHP_EOL
+        );
+
+        return $addedMediaBundleAsset->getResourceName();
+    }
+
+    /**
+     * Creates a new HTML5 display upload ad and adds it to the specified ad group.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID where the new ad will be added to
+     * @param string $adAssetResourceName the resource name of the media bundle containing the
+     *     HTML5 components
+     */
+    private static function createDisplayUploadAdGroupAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        string $adAssetResourceName
+    ) {
+        // Creates an ad group ad for the new ad.
+        $adGroupAd = new AdGroupAd([
+            'ad' => new Ad([
+                'name' => new StringValue(['value' => 'Ad for HTML5']),
+                'final_urls' => [new StringValue(['value' => 'http://example.com/html5'])],
+                // Exactly one ad data field must be included to specify the ad type. See
+                // https://developers.google.com/google-ads/api/reference/rpc/v3/Ad for the full
+                // list of available types.
+                'display_upload_ad' => new DisplayUploadAdInfo([
+                    'display_upload_product_type' => DisplayUploadProductType::HTML5_UPLOAD_AD,
+                    'media_bundle' => new AdMediaBundleAsset([
+                        'asset' => new StringValue(['value' => $adAssetResourceName])
+                    ])
+                ])
+            ]),
+            'ad_group' => ResourceNames::forAdGroup($customerId, $adGroupId),
+            'status' => AdGroupAdStatus::PAUSED
+        ]);
+
+        // Creates an ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+
+        // Issues a mutate request to add the ad group ads.
```

Comment:
"ads" to "ad"?

https://github.com/googleads/google-ads-php/blob/master/examples/BasicOperations/AddResponsiveSearchAd.php#L152